### DISCUSSION TITLE
Unable to get the correct version detected

### DISCUSSION THREAD

**Discussion by mortenn** [PaulHatch/semantic-version#147] [Q&A] [answered]
Hi, I'm trying to set up this action in my new pipeline, but it keeps detecting the wrong version.
I am at a loss what is going wrong here, please help <3

https://github.com/mortenn/BrowserPicker/actions/runs/8499864106/job/23281113301

```
Set output debug_output = {
  "commands": [
    {"command":"git","args":["rev-parse","--abbrev-ref","HEAD"],"output":"main\n","stderr":"","error":null},
    {"command":"git","args":["rev-list","-n1","--all"],"output":"87b65cea6053cea017aec6ca4f0d8e2379a6e2ad\n","stderr":"","error":null},
    {"command":"git","args":["rev-parse","HEAD"],"output":"87b65cea6053cea017aec6ca4f0d8e2379a6e2ad\n","stderr":"","error":null},
    {"command":"git tag --points-at 87b65cea6053cea017aec6ca4f0d8e2379a6e2ad v*[0-9].*[0-9].*[0-9]","args":[],"output":"","stderr":"","error":null},
    {"command":"git for-each-ref --sort=-v:*refname --format=%(refname:short) --merged=87b65cea6053cea017aec6ca4f0d8e2379a6e2ad refs/tags/v*[0-9].*[0-9].*[0-9]","args":[],"output":"v2.0.0-beta4\nv2.0-beta3.1\nv1.8.2\nv1.8.1\nv1.8.0\nv1.7.1\nv1.7.0\n","stderr":"","error":null},
    {"command":"git","args":["merge-base","v1.8.2","87b65cea6053cea017aec6ca4f0d8e2379a6e2ad"],"output":"f544fa51cc7f52857134307e7359c9a5bbfb5160\n","stderr":"","error":null},
    {"command":"git log --pretty
```
As we can see, it finds the v2.0.0-beta4 tag, but ignores that.

I also tried doing a commit to main with "(MAJOR)" in the message, but it still showed up as 1.8.3:
```
Version is 1.8.3-beta54
```

I can probably work around this by adding a v2.0.0 tag I guess, but I haven't released it yet, so I don't really want to do that unless there is absolutely no way around the problem.

**Comment by PaulHatch** [comment, marked as answer]
The change path basically adds a `-- ${CHANGE_PATH}` to all git commands is intended for fully independently versioning a sub project when you have multiple projects present and is typically used along with a `namespace` param. With this enabled, only commits on that path are considered. You have a commit with `(MAJOR)` which changes the readme, but this file isn't in the change path which is given as `/src` so it won't be considered.

If you haven't seen this already, you might want to have a look at the [guide](https://github.com/PaulHatch/semantic-version/blob/master/guide.md) for details on common configurations.

**Reply by mortenn** [reply]
excellent, thanks, now I get something more like expected;
`##[debug]Set output version = 2.0.0-beta3`

I did want beta4, but that'll come once I make another commit I guess.

now, I am building dotnet here, which means I would like version_prefix=`2.0.0` and version_suffix=`beta3`, but it seems I would need to either parse the version string or put together `major`, `minor` and `patch` with `"beta"` and `increment` myself. Am I missing something?
